src/aml/bert.py

```diff
@@ -0,0 +1,203 @@
+            for d in preprocessed_test:
+                file.write(d + '\n')
+
+        pd.to_pickle(labels, f'{path}/test-labels.pk')
+#--------------------------------------------------------------------------------------------------
+# Class Definition
+#--------------------------------------------------------------------------------------------------
+
+# TODO: Change these
+# @inproceedings{DBLP:conf/www/YanGLC13,
+#   author       = {Xiaohui Yan and Jiafeng Guo and Yanyan Lan and Xueqi Cheng},
+#   title        = {A biterm topic model for short texts},
+#   booktitle    = {22nd International World Wide Web Conference, {WWW} '13, Rio de Janeiro, Brazil, May 13-17, 2013},
+#   pages        = {1445--1456},
+#   publisher    = {International World Wide Web Conferences Steering Committee / {ACM}},
+#   year         = {2013},
+#   url          = {https://doi.org/10.1145/2488388.2488514},
+#   biburl       = {https://dblp.org/rec/conf/www/YanGLC13.bib},
+# }
+class BERT(AbstractAspectModel):
+    def __init__(self, naspects, nwords): 
+        super().__init__(naspects, nwords)
+
+    # TODO: What is this? should I change it?
+    def load(self, path):
+        self.mdl = pd.read_pickle(f'{path}model')
+        assert self.mdl.topics_num_ == self.naspects
+        self.dict = pd.read_pickle(f'{path}model.dict')
+    
+    def train(self,
+              reviews_train: List[Review],
+              reviews_valid: Optional[List[Review]],
+              am: str,
+              doctype: Optional[str],
+              no_extremes: Optional[bool],
+              output: str
+    ):
+        try:
+            save_train_reviews_to_file(reviews_train, output)
+
+            args = {
+            'model_type': 'bert',
+            'absa_type': 'tfm',
+            'tfm_mode': 'finetune',
+            'fix_tfm': 0,
+            'model_name_or_path': 'bert-base-uncased',
+            'data_dir': output,
+            'task_name': 'lady',
+            'per_gpu_train_batch_size': 16,
+            'per_gpu_eval_batch_size': 8,
+            'learning_rate': 2e-5,
+            'do_train': True,
+            'do_eval': True,
+            'do_lower_case': True,
+            'tagging_schema': 'BIEOS',
+            'overfit': 0,
+            'overwrite_output_dir': True,
+            'eval_all_checkpoints': True,
+            'MASTER_ADDR': 'localhost',
+            'MASTER_PORT': 28512,
+            'max_steps': 1500
+            }
+
+            model = train.main(Namespace(**args))
+
+            pd.to_pickle(model, f'{output}model')
+        except Exception as e:
+            raise RuntimeError(f'Error in training BERT model: {e}')
+
+    def infer_batch(self, reviews_test, h_ratio, doctype, output):
+        save_test_reviews_to_file(reviews_test, h_ratio, output)
+
+        absa_home = 'CHANGE_ME'
+        args = {
+            # TODO: change this to the model directory
+            'absa_home': f'{absa_home}',
+            'ckpt': f'{absa_home}/checkpoint-1200',
```

Comment:
@3ripleM are we going to fix this?

Reply:
@hosseinfani  Sure, we are not done yet. I just made a draft PR for followups